### DISCUSSION TITLE
`HOME` env not set in pods for certain deployments (minio/operator deployed through Helm)

### DISCUSSION THREAD

**Discussion by privatecoder** [syself/cluster-api-provider-hetzner#396] [Q&A] [answered]
Hi @batistein, hi @janiskemper

I have another question regarding a weird issue:

I have three clusters, a K3s- (v1.24.4), a managed K8s- (v1.23.12) and a new CAPH K8s-cluster (v1.24.7).

When deploying the minio/operator Helm-Chart (any version) to all three of them (individually), it does not run on the CAPH K8s-cluster (v1.24.7) because the pods of its deployment are missing the `HOME` env-var (i.e. `echo $HOME` returns nothing).

_Regular_ deplyoments and pods are not affected.

I would say this is a minio-issue but the chart does work on the other two clusters (where the `HOME` env-var is exposed, i.e. `echo $HOME` returns `/root`).

ANY ideas?

**Comment by privatecoder** [comment]
Any thoughts on this one? @batistein? I create another cluster (HCloud only) with the same result.

**Reply by batistein** [reply]
pretty tough to tell from this. I' d guess that this might have something to do with the container runtime. We use in the example template containerd with crun.

**Reply by batistein** [reply]
I think i got it :D https://github.com/containers/crun/issues/108

**Reply by batistein** [reply]
but should be already fixed... No idea what it could have triggered then. But I think the direction is the right one.

**Reply by privatecoder** [reply]
wow, that was quick – I'll have a look and report back!

**Reply by privatecoder** [reply]
it was closed in 2019 however it's still not fixed in 2021 according to this [crun/pull/599](https://github.com/containers/crun/pull/599), this [cilium/issues/14301](https://github.com/cilium/cilium/issues/14301) and this [podman/issues/9378](https://github.com/containers/podman/issues/9378).. and others..

**Reply by batistein** [reply]
ok you can easily test it out by creating a [runtimeClass](https://kubernetes.io/docs/concepts/containers/runtime-class/) object pointing to runc. This is also already installed. And then reference it with runtimeClassName in your deployment.

**Reply by batistein** [reply]
or you can remove crun; or change the default runtime. See: https://github.com/syself/cluster-api-provider-hetzner/blob/main/templates/cluster-templates/bases/kct-md-0-ubuntu.yaml#L39-L50

**Reply by privatecoder** [reply]
I will give that a try tomorrow – found the "try runc" tip in [a podman issue thread](https://github.com/containers/podman/issues/9378) as well. crun seems to be the lightest and quickest solution though and MinIO should ensure a non-empty $HOME var if it requires it.. I opened an issue there as well :)

**Comment by privatecoder** [comment, marked as answer]
So I did modify the MinIO Helm chart locally to enable passing `runtimeClassName` as parameter and deploying it after creating a `RuntimeClass`-object for `runc` as suggested.

It **_did_** solve the issue; `$HOME` indeed is present with `runc` and MinIO does not crash anymore..

**Reply by privatecoder** [reply]
@batistein just as a final note here: Maybe using `runc` would be the better default-choice for the the template as it seems to be less _picky_ in certain scenarios like this one and people who are _more familiar with containerd runtime plugins_ can still easily change it back to `crun`.

Anyway: Thank you again for your support!
